### ISSUE TITLE
Discussion : Migrate to PM2

### ISSUE THREAD

**Issue by YashKumarVerma** [camicroscope/Caracal#94]
**Is your feature request related to a problem? Please describe.**
- the code currently relies on [throng](https://www.npmjs.com/package/throng), 
- throng handles a subset of features that are provided by other alternatives like [pm2](https://pm2.keymetrics.io/)
- and most importantly, requires a **particular application structure** and code refactor to work [like this function](https://github.com/camicroscope/Caracal/blob/master/caracal.js#L174-L197)

## New Possibilities that open up
- use a watch mode to test changes in realtime
- completely isolate application logic and cluster configurations
- dump logs to file

**Comment by birm** [comment]
I'm not 100% sold, but watching configs could be a good addition. It's got to be better than nodemon, right? :)

**Comment by YashKumarVerma** [comment]
yeah nodemon could definitely handle the live update part, and since nodemon supports a custom config file, we can properly document the configs too.

**Comment by birm** [comment]
That's fair. So long as it's not too big/slow on install, I see no issue. :)
